docs/BackendAPI.md

```diff
@@ -0,0 +1,52 @@
+# ONNX Backend Interface Proposal
+
+We propose a cross-platform API for loading and executing ONNX graphs on optimized backends. High-level frameworks and applications can use this API to execute neural network and machine learning models. Hardware vendors can implement this API to expose specialized hardware accelerators and highly optimized software infrastructure to the users.
+
+## Core Features
+
+- Standardized interface for neural network inference on special-purpose accelerators, CPUs, GPUs, DSPs, and FPGAs
+- Based on widely supported technologies
+  - C API for function calls
+  - ONNX format for passing model graphs
+  - NCHW tensor layout for passing inputs and outputs
+- Dynamic discovery of available backends for model execution
+  - Multiple backends from different vendors can co-exist on the same system
+- Dynamic discovery of supported ONNX Operators on each backend
+
+### Optional Features:
+
+- Graphs with variable-shape inputs and/or outputs
+- Graphs with data-dependendent output shapes
+
+## How to Use ONNX Backend Interface
+
+0. (Optional) Use `onnx_backend_load` to dynamically load the ONNX Backend Interface library.
```

Comment:
Why this is optional ?
BE implementation can be built with BE API in the same lib. ?

Reply:
To clarify: BE implementation is a software library that implements the 9 functions defined in the backend API. A typical user of a BE API is a deep learning library.

This step is optional for two reasons:
1. BE loader library (which provides `onnx_backend_load`) is not a part of the spec, it is provided only for convenience. Deep learning frameworks can use their own loader code instead.
2. In embedded environments, where the Backend API library is always available, users can use dynamic linking instead of dynamic loading (i.e. pass `-lonnxbe` argument at link time instead of using `dlopen`).